lgc/test/Transforms/CpsLowering/cps-unify-exits.lgc

```diff
@@ -0,0 +1,202 @@
+; NOTE: Assertions have been autogenerated by tool/update_llpc_test_checks.py UTC_ARGS: --tool lgc --function-signature
+; RUN: lgc -mcpu=gfx1030 -o - -passes='require<lgc-pipeline-state>,lgc-patch-entry-point-mutate' %s | FileCheck --check-prefixes=CHECK %s
+
+declare void @lgc.cps.jump(...) noreturn
+
+define void @unify_jumps({i32} %state, i32 %arg, ptr %table) !lgc.cps !0 {
+entry:
+  %v = extractvalue {i32} %state, 0
+  %cond = icmp ult i32 %v, 3
+  br i1 %cond, label %then, label %else
+
+then:
+  %table.0 = getelementptr i32, ptr %table, i32 0
+  %cr.then = load i32, ptr %table.0
+  %then.arg = add i32 %arg, 1
+  %v.then = mul i32 %v, 2
+  %state.then = insertvalue {i32} poison, i32 %v.then, 0
+  call void (...) @lgc.cps.jump(i32 %cr.then, i32 2, {i32} %state.then, i32 %then.arg)
+  unreachable
+
+else:
+  %table.1 = getelementptr i32, ptr %table, i32 1
+  %cr.else = load i32, ptr %table.1
+  %else.arg = uitofp i32 %arg to float
+  call void (...) @lgc.cps.jump(i32 %cr.else, i32 2, {} poison, float %else.arg, i32 5)
+  unreachable
+}
+
+define void @unify_jump_ret({i32} %state, i32 %arg, ptr %table) !lgc.cps !0 {
+entry:
+  %v = extractvalue {i32} %state, 0
+  %cond = icmp ult i32 %v, 3
+  br i1 %cond, label %then, label %else
+
+then:
+  %table.0 = getelementptr i32, ptr %table, i32 0
+  %cr.then = load i32, ptr %table.0
+  %then.arg = add i32 %arg, 1
+  %v.then = mul i32 %v, 2
+  %state.then = insertvalue {i32} poison, i32 %v.then, 0
+  call void (...) @lgc.cps.jump(i32 %cr.then, i32 2, {i32} %state.then, i32 %then.arg)
+  unreachable
+
+else:
+  ret void
+}
+
+!0 = !{i32 1} ; level 1
+; CHECK-LABEL: define {{[^@]+}}@unify_jumps.1
```

Comment:
Not part of this patch but the cps lowering should call `takeName` and others (especially `insertAfter` for function order) that are also called in `lgc::addFuncitonArgs`.

Reply:
Yes, agree the takeName should be there. Do we need function order for particular reason or just make reading print-after log easier? Others should be covered by `copyMetadata()` and `copyAttributesFrom()`.

Reply:
I think it’s helpful to preserve function order when writing tests with multiple functions (e.g. it allows auto-generating function-CHECKs).
So if it doesn’t hurt, I would add it :)

Reply:
Fixed in new commit. Thanks!